File: experiments/llm-reasoning-cost/generated_code/cpp/05_file_io_codex.cpp
#include <fstream>
#include <string>

enum class WriteError {
    None,
    OpenFailed,
    WriteFailed,
    CloseFailed
};

struct WriteResult {
    bool ok;
    WriteError error;
};

WriteResult write_greeting(const std::string& filename) {
    std::ofstream out(filename, std::ios::out | std::ios::trunc);
    if (!out.is_open()) {
        return {false, WriteError::OpenFailed};
    }

    out << "Hello from the program!";
    if (!out) {
        return {false, WriteError::WriteFailed};
    }

    out.close();
    if (!out) {
        return {false, WriteError::CloseFailed};
    }

    return {true, WriteError::None};
}

File: experiments/llm-reasoning-cost/generated_code/go/05_file_io_codex.go
package main

import (
	"os"
)

func write_greeting(filename string) error {
	f, err := os.Create(filename)
	if err != nil {
		return err
	}
	defer f.Close()

	if _, err := f.WriteString("Hello from the program!"); err != nil {
		return err
	}

	return nil
}
